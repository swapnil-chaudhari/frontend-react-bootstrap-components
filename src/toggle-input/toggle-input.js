import PropTypes from 'prop-types';
import React from 'react';
import cx from 'classnames';
import './toggle-input.scss';


export const ToggleInput = ({
    qaid,
    className,
    leftLabel,
    rightLabel,
    leftValue,
    rightValue,
    value,
    onClick
}) => {
    const mainClassName = cx('ToggleInput', className);

    let left = false;
    if (leftValue === value)
        left = true;

    const toggleClassName = cx('ToggleInput-toggle', {
        'ToggleInput-toggle-left': left,
        'ToggleInput-toggle-right': !left,
    });

    const labelLeftClassName = cx('ToggleInput-label', {
        'ToggleInput-label-enabled': left
    });

    const labelRightClassName = cx('ToggleInput-label', {
        'ToggleInput-label-enabled': !left
    });

    let toggledValue = leftValue;
    if (leftValue === value)
        toggledValue = rightValue;

    return (
        <div id={ qaid } className={ mainClassName }>
            <div className={ labelLeftClassName } onClick={ () => onClick(leftValue) }>
                { leftLabel }
            </div>
            <div className="ToggleInput-toggle-bg" onClick={ () => onClick(toggledValue) }>
                <div className={ toggleClassName } />
            </div>
            <div className={ labelRightClassName } onClick={ () => onClick(rightValue) }>
                { rightLabel }
            </div>
        </div>
    );
};

ToggleInput.displayName = 'ToggleInput';

ToggleInput.propTypes = {
    leftLabel: PropTypes.string,
    rightLabel: PropTypes.string,
    leftValue: PropTypes.any.isRequired,
    rightValue: PropTypes.any.isRequired,
    qaid: PropTypes.string,
    className: PropTypes.string,
    value: PropTypes.any.isRequired,
    onClick: PropTypes.func.isRequired,
};

export default ToggleInput;
