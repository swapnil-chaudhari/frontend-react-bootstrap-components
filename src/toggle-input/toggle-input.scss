@import "src/styles/variables";

.ToggleInput {
    &-label {
        color: $timberwolf;
        cursor: pointer;
        display: inline-block;
        font-size: 9px;
        font-weight: 500;
        letter-spacing: .3px;
        transition: color .2s ease-in-out;
        vertical-align: middle;
    }

    &-label-enabled {
        color: $bluefish;
    }

    &-toggle-bg {
        background-color: $bluefish;
        border-radius: 20px;
        cursor: pointer;
        display: inline-block;
        height: 18px;
        margin-left: 8px;
        margin-right: 8px;
        position: relative;
        vertical-align: middle;
        width: 30px;
    }

    &-toggle {
        background-color: white;
        border-radius: 7px;
        height: 14px;
        position: absolute;
        top: 2px;
        transition: left .2s ease-in-out;
        width: 14px;
    }

    &-toggle-left {
        left: 3px;
    }

    &-toggle-right {
        left: 13px;
    }
}
