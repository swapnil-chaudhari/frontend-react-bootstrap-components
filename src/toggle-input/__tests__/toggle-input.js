import { shallow } from 'enzyme';
import React from 'react';
import { ToggleInput } from '../toggle-input';

describe('ToggleInput', () => {
    let props;

    beforeEach(() => {
        props = {
            leftLabel: 'Yes',
            rightLabel: 'No',
            leftValue: false,
            rightValue: true,
            value: true,
            onClick: jest.fn(),
            qaid: 'qaid',
            className: 'className'
        };
    });

    it('renders the toggle input', () => {
        const component = shallow(<ToggleInput { ...props } />);
        expect(component).toMatchSnapshot();
    });

    it('toggles', () => {
        const component = shallow(<ToggleInput { ...props } />);

        component.find('.ToggleInput-toggle-bg').first().props().onClick();
        expect(props.onClick).toHaveBeenCalledWith(false);
    });

    it('will be the left if you click left label', () => {
        const component = shallow(<ToggleInput { ...props } />);

        component.find('.ToggleInput-label').first().props().onClick();
        expect(props.onClick).toHaveBeenCalledWith(false);
    });

    it('will be the right if you click right label', () => {
        const component = shallow(<ToggleInput { ...props } />);

        component.find('.ToggleInput-label').last().props().onClick();
        expect(props.onClick).toHaveBeenCalledWith(true);
    });

    it('will change classnames when on the left', () => {
        props.value = false;
        const component = shallow(<ToggleInput { ...props } />);

        expect(component.find('.ToggleInput-label').first())
            .toHaveClassName('ToggleInput-label-enabled');
        expect(component.find('.ToggleInput-toggle').first())
            .toHaveClassName('ToggleInput-toggle-left');
    });
});
