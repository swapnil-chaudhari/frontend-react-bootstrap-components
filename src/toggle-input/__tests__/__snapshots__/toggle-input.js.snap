// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToggleInput renders the toggle input 1`] = `
<div
  className="ToggleInput className"
  id="qaid"
>
  <div
    className="ToggleInput-label"
    onClick={[Function]}
  >
    Yes
  </div>
  <div
    className="ToggleInput-toggle-bg"
    onClick={[Function]}
  >
    <div
      className="ToggleInput-toggle ToggleInput-toggle-right"
    />
  </div>
  <div
    className="ToggleInput-label ToggleInput-label-enabled"
    onClick={[Function]}
  >
    No
  </div>
</div>
`;
